messaging/src/main/protos/Model.proto

```diff
@@ -20,11 +20,23 @@ message ContactId {
   string id = 2; // the public IdentityKey for direct contacts, the globally unique group id (type 4 UUID) for groups, both base32 encoded
 }
 
+enum ContactSource {
+  UNKNOWN = 0; // source of contact is unknown
+  INTRODUCTION = 1; // contact was added through introduction
+  APP1 = 2; // application-specific source 1
```

Comment:
messaging-android doesn't know or care what these are, the application can define its own semantics for which source is which.

Reply:
I think you added that, so we do default to UNKNOWN now :)